### PR TITLE
#162375483 Implement upload progress bar and file type check

### PR DESCRIPTION
**Pull request by janetnim**
#### What does this PR do?
Displays a progress bar on guesthouse image upload and sends an error message pop up when the image file type is not jpeg or png

#### Description of Task to be completed?
- Check that only jpeg or png files can be uploaded as guesthouse image
- Add a progress bar on image file upload

#### How should this be manually tested?
* Clone the repositories:
`git clone https://github.com/andela/travel_tool_front.git`
`git clone https://github.com/andela/travel_tool_back.git`
* Checkout to `develop branch` on travel_tool_back and `ft-guest-house-image-progressbar-162375483` in travel_tool_front:
`git checkout develop`
`git checkout ft-guest-house-image-progressbar-162375483`
* Install dependencies and start both projects with the commands:
`yarn install`
`make start`
* As a Travel Admin click manage under residence and add a guestroom
* Upload guest house image

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[162375483](https://www.pivotaltracker.com/story/show/162375483)

#### Screenshots (if appropriate)

gh2

gh1

#### Questions:
N/A

### CODE REVIEW

**Comment by janetnim** [comment]
@ronaldndirangu The image is only uploaded on submit(when you click the save button). So when you click upload what you see is a preview but the image has not yet been uploaded and the progress bar can only appear when the image is uploading.

**Comment by ronaldndirangu** [comment]
@janetnim Should the button read upload then or select?

**Comment by janetnim** [comment]
@ronaldndirangu I would have to seek more clarification on this from @seunzone

**Comment by ronaldndirangu** [comment]
@janetnim Cool update me on what he says.

File: src/components/Forms/NewAccommodationForm/NewAccommodation.jsx [hunk 1]
```diff
@@ -80,17 +81,26 @@ class NewAccommodation extends PureComponent {
     const reader = new FileReader();
     if (event.target.files[0]) {
       const file = event.target.files[0];
-      reader.onloadend = () => {
-        this.setState(
-          prevState => ({
-            values: { ...prevState.values, preview: reader.result, image: file, }
-          }), this.validate );}; reader.readAsDataURL(file);
+      if (file.type != ('image/jpeg' || 'image/png')) {
```
Comment (CEOehis) [inline src/components/Forms/NewAccommodationForm/NewAccommodation.jsx]: This check won't let PNG's be uploaded. I think the checks should be two separate checks.

Comment (CEOehis) [inline src/components/Forms/NewAccommodationForm/NewAccommodation.jsx]: This check won't let PNG's be uploaded. I think you should have two separate checks for both file types.

Comment (janetnim) [reply src/components/Forms/NewAccommodationForm/NewAccommodation.jsx]: Thanks for noting that. I have made the changes to check on that.

**Comment by samuelkubai** [comment]
@janetnim were the changes requested by Mathew implemented?

**Comment by janetnim** [comment]
@samuelkubai Am working on them

**Comment by janetnim** [comment]
@matthewacha In the upload callback function, I call setState which can only be done in the context of the stateful component. Creating an action creator for it will be difficult because the component `NewAccommodation` is not connected to Redux. I could connect it to redux and move the function around but then it will become unreadable and more complicated.

**Comment by samuelkubai** [comment]
@janetnim I am reviewing your commits and the commit does not conform to the playbook.

**Comment by janetnim** [comment]
@samuelkubai This must have happened when I was rebasing. Thanks for the heads up.

**Pull request merged**